autotest/autotest.py

```diff
@@ -94,16 +110,43 @@ def ansi(code):
 
 keymaps = { "^C": "\x03" }
 
-while True:
+wait_for, stop_action, method = "", "", ""
+
+interactive = True
+if os.environ.get("WORKSHOP_TEST_FORCE_NONINTERACTIVE"):
+    interactive=False
+
+while i < len(actions):
     with open("nextstep","w") as f:
         f.write(str(i))
     slide, snippet, method, data = actions[i]
     data = data.strip()
+
+    # Look behind at the last slide to see if 'wait' or 'keys' was defined.
```

Comment:
Ohhhh 🤦‍♂️ 

I hadn't realized that it would be so trivial to look at the *next snippet*, since we're walking an indexed array! So we could have the `wait` snippet be *after* the `bash` snippet if we want. Good, good. Do you think it makes more sense to have it before or after, after all?

Reply:
After, I'd say. Addressed in latest commit.